### PR TITLE
TASK: Adjust to Flow 7.0

### LINKED ISSUE #57: Replace Component with a Middleware or remove completely?
With Flow 7 Components will no longer be supported. However, looking at the current code of the component (https://github.com/neos/setup/blob/master/Classes/Core/ConfigureRoutingComponent.php), it just forces the router to load the Setup Routes.yaml and will then run as long as the package exists?

### PR DESCRIPTION
**Pull request by albe**
Resolves #57

### CODE REVIEW

File: Classes/Core/RequestHandler.php [hunk 1]
```diff
@@ -64,16 +64,18 @@ public function getPriority()
     public function handleRequest()
     {
         $request = ServerRequest::fromGlobals();
+        // TODO: Required for b/c - can be removed with Flow 7
         $response = new \GuzzleHttp\Psr7\Response();
         $this->componentContext = new ComponentContext($request, $response);
+        $this->bootstrap->setEarlyInstance(ComponentContext::class, $this->componentContext);
```
Comment (albe) [inline Classes/Core/RequestHandler.php:70]: Note: Those lines can be removed again with https://github.com/neos/flow-development-collection/pull/2221

File: Classes/Core/RequestHandler.php [hunk 2]
```diff
@@ -121,12 +123,12 @@ protected function checkBasicRequirementsAndDisplayLoadingScreen()
     protected function resolveDependencies()
     {
         $objectManager = $this->bootstrap->getObjectManager();
-        $componentChainFactory = $objectManager->get(ComponentChainFactory::class);
         $configurationManager = $objectManager->get(ConfigurationManager::class);
         $this->settings = $configurationManager->getConfiguration(ConfigurationManager::CONFIGURATION_TYPE_SETTINGS, 'Neos.Flow');
         $setupSettings = $configurationManager->getConfiguration(ConfigurationManager::CONFIGURATION_TYPE_SETTINGS, 'Neos.Setup');
-        $httpChainSettings = Arrays::arrayMergeRecursiveOverrule($this->settings['http']['chain'], $setupSettings['http']['chain']);
-        $this->baseComponentChain = $componentChainFactory->create($httpChainSettings);
+        $httpChainSettings = Arrays::arrayMergeRecursiveOverrule($this->settings['http']['middlewares'], $setupSettings['http']['middlewares']);
```
Comment (albe) [inline Classes/Core/RequestHandler.php:129]: Not really sure why that manual config merging shenanigan is necessary at all?

File: composer.json [hunk 3]
```diff
@@ -18,7 +18,7 @@
     },
     "extra": {
         "branch-alias": {
-            "dev-master": "5.0.x-dev"
+            "dev-master": "6.0.x-dev"
```
Comment (bwaidelich) [inline composer.json:21]: what is this?

Comment (kdambekalns) [reply composer.json:21]: Makes the `master` branch available as `6.0.x-dev` as well.

Comment (bwaidelich) [reply composer.json:21]: OK, but what is the context? And sorry, if that's a stupid question

File: Classes/Core/RequestHandler.php [hunk 4]
```diff
@@ -64,16 +64,18 @@ public function getPriority()
     public function handleRequest()
     {
         $request = ServerRequest::fromGlobals();
+        // TODO: Required for b/c - can be removed with Flow 7
         $response = new \GuzzleHttp\Psr7\Response();
         $this->componentContext = new ComponentContext($request, $response);
```
Comment (bwaidelich) [inline Classes/Core/RequestHandler.php:69]: This line will fail with Flow 7.0

File: composer.json (hunk 3 above)
Comment (kdambekalns) [reply composer.json:21]: Basically it is used to tell what versions are cut from a branch. While that is clear on `1.0` or `5.3`, it isn't so much on `master`.

That being said: Why did you raise this here, @albe? The current versions for `neos/setup` are 4.x, so 5.x seems fine, if it is to be expected that 5.x will be created on `master` soonish. But 6.x means skipping one… 🤔

Comment (bwaidelich) [reply composer.json:21]: yes, that's what I meant with context :)

File: Classes/Core/RequestHandler.php (hunk 2 above)
Comment (bwaidelich) [reply Classes/Core/RequestHandler.php:128]: mh, I don't think that this is still needed actually

**Comment by bwaidelich** [comment]
The setup still won't completely work because the session is not initialized for the Ajax widget to work. This is fixed with https://github.com/neos/flow-development-collection/pull/2261
Also I wonder if the `branch-alias` change is required (see comments)

File: composer.json (hunk 3 above)
Comment (albe) [reply composer.json:21]: I might have been a bit to eager on this one :)
```suggestion
            "dev-master": "5.0.x-dev"
```

**Pull request merged**